packages/auth/src/clients/service.test.ts

```diff
@@ -0,0 +1,332 @@
+import nock from 'nock'
+import { createTestApp, TestContainer } from '../tests/app'
+import { Config } from '../config/app'
+import { IocContract } from '@adonisjs/fold'
+import { initIocContainer } from '../'
+import { AppServices } from '../app'
+import { ClientService } from './service'
+
+const KEY_REGISTRY_ORIGIN = 'https://openpayments.network'
+const TEST_CLIENT_DISPLAY = {
+  name: 'Test Client',
+  uri: 'https://example.com'
+}
+
+const TEST_KID_PATH = '/keys/test-key'
+const TEST_CLIENT_KEY = {
+  kid: KEY_REGISTRY_ORIGIN + TEST_KID_PATH,
+  x: 'test-public-key',
+  kty: 'OKP',
+  alg: 'EdDSA',
+  crv: 'Ed25519',
+  key_ops: ['sign', 'verify'],
+  use: 'sig'
+}
+
+describe('Client Service', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let clientService: ClientService
+
+  beforeAll(
+    async (): Promise<void> => {
+      deps = await initIocContainer(Config)
+      clientService = await deps.use('clientService')
+      appContainer = await createTestApp(deps)
+    }
+  )
+
+  afterAll(
+    async (): Promise<void> => {
+      nock.restore()
+      await appContainer.shutdown()
+    }
+  )
+
+  describe('Registry Validation', (): void => {
+    describe('Client Properties', (): void => {
+      test('Can validate client properties with registry', async (): Promise<void> => {
+        const expDate = new Date()
+        expDate.setTime(expDate.getTime() + 1000 * 60 * 60)
+
+        const nbfDate = new Date()
+        nbfDate.setTime(nbfDate.getTime() - 1000 * 60 * 60)
+        nock(KEY_REGISTRY_ORIGIN)
+          .get('/keys/correct')
+          .reply(200, {
+            display: TEST_CLIENT_DISPLAY,
+            keys: {
+              ...TEST_CLIENT_KEY,
+              kid: KEY_REGISTRY_ORIGIN + '/keys/correct',
+              exp: Math.round(expDate.getTime() / 1000),
+              nbf: Math.round(nbfDate.getTime() / 1000),
+              revoked: false
+            }
+          })
+
+        const validClient = await clientService.validateClientWithRegistry({
+          display: TEST_CLIENT_DISPLAY,
+          key: {
+            proof: 'httpsig',
+            jwk: {
+              ...TEST_CLIENT_KEY,
+              kid: KEY_REGISTRY_ORIGIN + '/keys/correct'
+            }
+          }
+        })
+
+        expect(validClient).toEqual(true)
```

Comment:
You could also call [`isDone`](https://github.com/nock/nock#isdone) on the nock scopes.